src/Model/Resolver/Products/SearchCriteria/CollectionProcessor/FilterProcessor/ConfigurableProductAttributeFilter.php

```diff
@@ -86,4 +88,19 @@ public function apply(Filter $filter, AbstractDb $collection)
 
         return true;
     }
+
+    /**
+     * Save active filters to registry
+     *
+     * @param $attributeName
+     * @param $attributeValues
+     */
+    protected function _saveActiveFiltersToRegistry($attributeName, $attributeValues)
+    {
+        $registry = $this->registry->registry('filter_attributes') ?? [];
+        $registry[$attributeName] = $attributeValues;
+
+        $this->registry->unregister('filter_attributes');
+        $this->registry->register('filter_attributes', $registry);
+    }
```

Comment:
Registry is deprecated class and must not be used. There is a reason we are reading from it within this class (there is legacy for category), but new functionality must not use deprecated.
Especially in GraphQl, where there are circular calls and deferred code execution.